lib/autocomplete-manager.coffee

```diff
@@ -110,26 +113,59 @@ class AutocompleteManager
     return if @isCurrentFileBlackListed()
     cursor = @editor.getLastCursor()
     return unless cursor?
-    cursorPosition = cursor.getBufferPosition()
-    currentScope = cursor.getScopeDescriptor()
-    return unless currentScope?
-    currentScopeChain = currentScope.getScopeChain()
-    return unless currentScopeChain?
-
-    options =
-      editor: @editor
-      buffer: @buffer
-      cursor: cursor
```

Comment:
A new PR would be nice. Please dont push that commit to this PR. 

I dont see why it would need the cursor. The cursor is a special purpose object. We only need the editor and position to generate a prefix. 

I am not fond of providing a random function in the options. It feels pretty dirty.